test/e2e/sync/sync.go

```diff
@@ -0,0 +1,449 @@
+	Cos    string `yaml:"cos"`
+	Object string `yaml:"object"`
+	Plan   string `yaml:"plan"`
+	Region string `yaml:"region"`
+}
+
+type Target []struct {
+	Bucket string `yaml:"bucket"`
+	Plan   string `yaml:"plan"`
+	Region string `yaml:"region"`
+}
+
+func runSyncCMD(args ...string) (int, string, string) {
+	args = append([]string{"image", "sync"}, args...)
+	return utils.RunCMD("pvsadm", args...)
+}
+
+func randomInt(min, max int) int {
+	return min + rand.Intn(max-min)
+}
+
+func generateRandomString(length int) string {
+	rand.Seed(time.Now().UnixNano())
+	bytes := make([]byte, length)
+	for i := 0; i < length; i++ {
+		bytes[i] = byte(randomInt(65, 91))
+	}
+	return string(bytes)
+}
+
+func generateStruct() Spec {
+
+	fmt.Println("STEP: Generating Struct")
+	spec := make(Spec, NoOfSources)
+
+	for src := 0; src < NoOfSources; src++ {
+		spec[src].Source.Bucket = "bucket-" + strings.ToLower(generateRandomString(6))
+		spec[src].Source.Cos = "COS-Test-" + generateRandomString(6)
+		spec[src].Source.Object = ""
+		spec[src].Source.Plan = PlanSlice[randomInt(0, len(PlanSlice))]
+		spec[src].Source.Region = RegionSlice[randomInt(0, len(RegionSlice))]
+		spec[src].Target = make(Target, NoOfTargetsPerSource)
+
+		for tgt := 0; tgt < NoOfTargetsPerSource; tgt++ {
+			spec[src].Target[tgt].Bucket = "bucket-" + strings.ToLower(generateRandomString(6))
+			spec[src].Target[tgt].Plan = PlanSlice[randomInt(0, len(PlanSlice))]
+			spec[src].Target[tgt].Region = RegionSlice[randomInt(0, len(RegionSlice))]
+		}
+	}
+	return spec
+}
+
+func createSpecFile(spec Spec) bool {
+
+	fmt.Println("STEP: Creating Spec file")
+
+	f, err := os.Create("spec.yaml")
+	if err != nil {
+		fmt.Println("ERROR: ", err)
+		return false
+	}
+	defer f.Close()
+
+	specString, merr := yaml.Marshal(&spec)
+	if merr != nil {
+		fmt.Println("ERROR: ", merr)
+		return false
+	}
+	_, err2 := f.WriteString(string(specString))
+	if err2 != nil {
+		fmt.Println("ERROR: ", err2)
+		return false
+	}
+
+	fmt.Println(string(specString))
+	return true
+}
+
+func createCOSInstances(spec Spec) bool {
+	fmt.Println("STEP: Creating COS instances")
+
+	bxCli, err := client.NewClientWithEnv(APIKey, client.DefaultEnv, Debug)
+	if err != nil {
+		fmt.Println("ERROR: ", err)
+		return false
+	}
+
+	resourceGroupQuery := management.ResourceGroupQuery{
+		AccountID: bxCli.User.Account,
+	}
+
+	resGrpList, err := bxCli.ResGroupAPI.List(&resourceGroupQuery)
+	if err != nil {
+		fmt.Println("ERROR: ", err)
+		return false
+	}
+
+	var resourceGroupNames []string
+	for _, resgrp := range resGrpList {
+		resourceGroupNames = append(resourceGroupNames, resgrp.Name)
+	}
+
+	for _, src := range spec {
+		fmt.Println("STEP: Creating COS instance", src.Cos)
+		_, err = bxCli.CreateServiceInstance(src.Cos, ServiceType, ServicePlan,
+			ResourceGrp, ResourceGroupAPIRegion)
+		if err != nil {
+			fmt.Println("ERROR: ", err)
+			return false
+		}
+	}
+
+	return true
+
+}
+
+func deleteCOSInstances(spec Spec) bool {
```

Comment:
again as I mentioned in the above this can be generic which just deletes the cos by supplied name

Reply:
Done